src/eval/operation.rs

```diff
@@ -2513,6 +2531,71 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                     ))
                 }
             }
+            BinaryOp::DictionaryAssume() => {
+                let (
+                    Closure {
+                        body: contract_term,
+                        env: contract_env,
+                    },
+                    _,
+                ) = self.stack.pop_arg(&self.cache).ok_or_else(|| {
+                    EvalError::NotEnoughArgs(3, String::from("dictionary_assume"), pos_op)
+                })?;
+
+                let lbl = match_sharedterm! {t1, with {
+                        Term::Lbl(lbl) => lbl
+                    } else return Err(EvalError::TypeError(
+                        String::from("Lbl"),
+                        String::from("dictionary_assume, 2nd argument"),
+                        fst_pos,
+                        RichTerm {
+                            term: t1,
+                            pos: pos1,
+                        }
+                    ))
+                };
+
+                match_sharedterm! {t2,
+                    with {
+                        Term::Record(record_data) => {
+                            let contract = contract_term.closurize(&mut self.cache, &mut env2, contract_env);
+                            let record_contract = Term::Record(RecordData{
+                                fields: record_data.fields.iter().map(|(key, value)| (*key, Field {
```

Comment:
The formatting is probably a bit off here. It's pretty annoying that `matched_sharedterm!` makes `cargo fmt` ignore this part. I don't know if that can be fixed in any way; in the meantime, a laborious but functional way is to replace the `matched_sharedterm` with a normal pattern matching, format, and turn it back to the macro.